In [2]:
import pandas as pd 

# Ohio Data

Testing with 1 county file.

In [4]:
raw = pd.read_csv('raw_data/ADAMS.txt', sep=',', header=0)

C:\Users\Ern\AppData\Local\Temp\ipykernel_13688\208357657.py:1: DtypeWarning: Columns (20,48,52,60,63,78,79,96,99,103,110) have mixed types. Specify dtype option on import or set low_memory=False.
  raw = pd.read_csv('raw_data/ADAMS.txt', sep=',', header=0)


In [6]:
raw.head()

SOS_VOTERID  COUNTY_NUMBER  COUNTY_ID LAST_NAME FIRST_NAME MIDDLE_NAME  \
0  OH0016294903              1      12125  KIMMERLY       TONY       LANCE   
1  OH0019991193              1      28536     BAKER     TRAVIS        COLE   
2  OH0026697487              1      40376  BURCHETT      JAMES       DAVID   
3  OH0016300071              1      19676     RILEY      DAVID           E   
4  OH0016299863              1       7611   ALLISON      TREVA         MAE   

  SUFFIX DATE_OF_BIRTH REGISTRATION_DATE  VOTER_STATUS  ...  \
0    NaN    1975-03-16        2020-04-28        ACTIVE  ...   
1    NaN    1987-12-20        2008-09-09  CONFIRMATION  ...   
2    NaN    1977-11-08        2022-10-12        ACTIVE  ...   
3    NaN    1981-02-18        1999-10-22        ACTIVE  ...   
4    NaN    1956-07-18        1988-11-23        ACTIVE  ...   

  GENERAL-11/03/2020 PRIMARY-05/04/2021 PRIMARY-08/03/2021 PRIMARY-09/14/2021  \
0                  X                NaN                NaN                NaN   
1                NaN                NaN                NaN                NaN   
2                NaN                NaN                NaN                NaN   
3                  X                NaN                NaN                NaN   
4                  X                NaN                NaN                NaN   

  GENERAL-11/02/2021  PRIMARY-05/03/2022  PRIMARY-08/02/2022  \
0                NaN                 NaN                 NaN   
1                NaN                 NaN                 NaN   
2                NaN                 NaN                 NaN   
3                  X                   R                 NaN   
4                  X                   R                 NaN   

   GENERAL-11/08/2022  SPECIAL-02/28/2023 PRIMARY-05/02/2023  
0                 NaN                 NaN                NaN  
1                 NaN                 NaN                NaN  
2                 NaN                 NaN                NaN  
3                   X                 NaN                NaN  
4                   X                 NaN                NaN  

[5 rows x 119 columns]

There are 119 columns. There is documentation on what the columns are and what the values mean.

In [9]:
col_list = raw.columns.values.tolist()

In [10]:
col_list

['SOS_VOTERID',
 'COUNTY_NUMBER',
 'COUNTY_ID',
 'LAST_NAME',
 'FIRST_NAME',
 'MIDDLE_NAME',
 'SUFFIX',
 'DATE_OF_BIRTH',
 'REGISTRATION_DATE',
 'VOTER_STATUS',
 'PARTY_AFFILIATION',
 'RESIDENTIAL_ADDRESS1',
 'RESIDENTIAL_SECONDARY_ADDR',
 'RESIDENTIAL_CITY',
 'RESIDENTIAL_STATE',
 'RESIDENTIAL_ZIP',
 'RESIDENTIAL_ZIP_PLUS4',
 'RESIDENTIAL_COUNTRY',
 'RESIDENTIAL_POSTALCODE',
 'MAILING_ADDRESS1',
 'MAILING_SECONDARY_ADDRESS',
 'MAILING_CITY',
 'MAILING_STATE',
 'MAILING_ZIP',
 'MAILING_ZIP_PLUS4',
 'MAILING_COUNTRY',
 'MAILING_POSTAL_CODE',
 'CAREER_CENTER',
 'CITY',
 'CITY_SCHOOL_DISTRICT',
 'COUNTY_COURT_DISTRICT',
 'CONGRESSIONAL_DISTRICT',
 'COURT_OF_APPEALS',
 'EDU_SERVICE_CENTER_DISTRICT',
 'EXEMPTED_VILL_SCHOOL_DISTRICT',
 'LIBRARY',
 'LOCAL_SCHOOL_DISTRICT',
 'MUNICIPAL_COURT_DISTRICT',
 'PRECINCT_NAME',
 'PRECINCT_CODE',
 'STATE_BOARD_OF_EDUCATION',
 'STATE_REPRESENTATIVE_DISTRICT',
 'STATE_SENATE_DISTRICT',
 'TOWNSHIP',
 'VILLAGE',
 'WARD',
 'PRIMARY-03/07/2000',
 'GENERAL-11

In [15]:
oh_adams_data = raw.copy()

In [20]:
len(oh_adams_data.columns)

119

In [31]:
# Get rid of historic data; just want voter info (at the moment)
last_col = oh_adams_data.columns.get_loc('WARD') # 45
cols_to_drop = oh_adams_data.columns[range(last_col+1,len(oh_adams_data.columns)-1)]
oh_adams = oh_adams_data.drop(cols_to_drop, axis=1)


In [32]:
oh_adams.head()

SOS_VOTERID  COUNTY_NUMBER  COUNTY_ID LAST_NAME FIRST_NAME MIDDLE_NAME  \
0  OH0016294903              1      12125  KIMMERLY       TONY       LANCE   
1  OH0019991193              1      28536     BAKER     TRAVIS        COLE   
2  OH0026697487              1      40376  BURCHETT      JAMES       DAVID   
3  OH0016300071              1      19676     RILEY      DAVID           E   
4  OH0016299863              1       7611   ALLISON      TREVA         MAE   

  SUFFIX DATE_OF_BIRTH REGISTRATION_DATE  VOTER_STATUS  ...  \
0    NaN    1975-03-16        2020-04-28        ACTIVE  ...   
1    NaN    1987-12-20        2008-09-09  CONFIRMATION  ...   
2    NaN    1977-11-08        2022-10-12        ACTIVE  ...   
3    NaN    1981-02-18        1999-10-22        ACTIVE  ...   
4    NaN    1956-07-18        1988-11-23        ACTIVE  ...   

  MUNICIPAL_COURT_DISTRICT        PRECINCT_NAME PRECINCT_CODE  \
0                      NaN     LIBERTY TOWNSHIP         01AAI   
1                      NaN  WINCHESTER TOWNSHIP         01ABG   
2                      NaN         WEST UNION A         01ABA   
3                      NaN         WEST UNION A         01ABA   
4                      NaN     LIBERTY TOWNSHIP         01AAI   

  STATE_BOARD_OF_EDUCATION STATE_REPRESENTATIVE_DISTRICT  \
0                        5                            90   
1                        5                            90   
2                        5                            90   
3                        5                            90   
4                        5                            90   

   STATE_SENATE_DISTRICT             TOWNSHIP             VILLAGE  WARD  \
0                     14     LIBERTY TOWNSHIP                 NaN   NaN   
1                     14  WINCHESTER TOWNSHIP                 NaN   NaN   
2                     14      TIFFIN TOWNSHIP  WEST UNION VILLAGE   NaN   
3                     14      TIFFIN TOWNSHIP  WEST UNION VILLAGE   NaN   
4                     14     LIBERTY TOWNSHIP                 NaN   NaN   

  PRIMARY-05/02/2023  
0                NaN  
1                NaN  
2                NaN  
3                NaN  
4                NaN  

[5 rows x 47 columns]